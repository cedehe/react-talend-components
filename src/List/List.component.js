import React, { PropTypes } from 'react';
import Toolbar from './Toolbar';
import Items from './Items';

/**
 * @param {object} props react props
 * @example
const props = {
	displayMode: 'table' / 'large' / 'tile' / component
	list: {
		items: [{}, {}, ...],
		columns: [
			{key, label},
			{key, label},
		]
	},
	toolbar: {
		display: {
			onChange: function,
		},
		sort: {
			field: key,
			options: [
				{key, label},
			],
			isDescending: true / false,
			onChange: function,
		},
		pagination: {
			onChange: function,
			itemsLength: number,
		},
		filter: {
			onFilter: function,
		},
	}
}
<List {...props}></List>
 */
function List({ id, displayMode, toolbar, list }) {
	let toolbarProps;
	if (toolbar) {
		toolbarProps = Object.assign({}, toolbar, { id });
		if (toolbar.display) {
			toolbarProps.display.mode = displayMode;
		}
		if (list.itemProps && list.itemProps.isSelected && list.itemProps.onToggleAll) {
			toolbarProps.selectAllCheckbox = {
				id,
				items: list.items,
				isSelected: list.itemProps.isSelected,
				onToggleAll: list.itemProps.onToggleAll,
			};
		}
	}
	return (
		<div className="tc-list">
			{toolbar && (<Toolbar {...toolbarProps} />)}
			<Items id={id} displayMode={displayMode} {...list} />
		</div>
	);
}

List.propTypes = {
	id: PropTypes.string,
	displayMode: PropTypes.string,
	list: PropTypes.shape(Items.propTypes),
	toolbar: PropTypes.shape(Toolbar.propTypes),
};

List.defaultProps = {
	displayMode: 'table',
};

export default List;
