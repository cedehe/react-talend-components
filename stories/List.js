import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { List, IconsProvider } from '../src/index';

const selected = [
	{
		id: 2,
		name: 'Foo',
		created: '2016-09-22',
		modified: '2016-09-22',
		author: 'Jean-Pierre DUPONT',
		icon: 'fa fa-file-pdf-o',
	}];
const props = {
	id: 'talend',
	displayMode: 'table',
	list: {
		titleProps: {
			key: 'name',
			iconKey: 'icon',
			displayModeKey: 'display',
			onClick: action('onClick'),
			onCancel: action('onCancel'),
			onChange: action('onChange'),
		},
		columns: [
			{ key: 'id', label: 'Id' },
			{ key: 'name', label: 'Name' },
			{ key: 'author', label: 'Author' },
			{ key: 'created', label: 'Created' },
			{ key: 'modified', label: 'Modified' },
		],
		items: [
			{
				id: 1,
				name: 'Title with actions',
				created: '2016-09-22',
				modified: '2016-09-22',
				author: 'Jean-Pierre DUPONT',
				actions: [{
					label: 'edit',
					icon: 'talend-edit',
					onClick: action('onEdit'),
				}, {
					label: 'delete',
					icon: 'talend-delete',
					onClick: action('onDelete'),
				}, {
					displayMode: 'dropdown',
					label: 'related items',
					icon: 'talend-folder',
					items: [
						{
							label: 'document 1',
							onClick: action('document 1 click'),
						},
						{
							label: 'document 2',
							onClick: action('document 2 click'),
						},
					],
				}],
				icon: 'fa fa-file-excel-o',
				display: 'text',
			},
			{
				id: 2,
				name: 'Title in input mode',
				created: '2016-09-22',
				modified: '2016-09-22',
				author: 'Jean-Pierre DUPONT',
				icon: 'fa fa-file-pdf-o',
				display: 'input',
			},
			{
				id: 2,
				name: 'Super long title to trigger overflow on tile rendering',
				created: '2016-09-22',
				modified: '2016-09-22',
				author: 'Jean-Pierre DUPONT with super long name',
			},
		],
		onElementSelect: action('onSelect'),
		onToggleAll: action('onToggleAll'),
		onToggleSingle: action('onToggleSingle'),
		ifSelected: (item) => {
			let found = false;
			for (let i = 0; i < selected.length; i += 1) {
				if (selected[i].id === item.id) {
					found = true;
					break;
				}
			}
			return found;
		},
	},
	toolbar: {
		onClickAdd: action('onClickAdd'),
		listActions: [
			{
				label: 'Delete selection',
				icon: 'talend-delete',
				onClick: action('delete'),
			},
		],
		onFilter: action('onFilter'),
		onSelectDisplayMode: action('onSelectDisplayMode'),
		sortBy: [
			{ id: 'id', name: 'Id' },
			{ id: 'name', name: 'Name', selected: true },
		],
		onSelectSortBy: action('onSelectSortBy'),
		itemsLength: 3,
		onChangePagination: action('onChangePagination'),
	},
};

storiesOf('List', module)
	.add('table (default)', () => (
		<div>
			<h1>List</h1>
			<h2>Definition</h2>
			<p>Display a list by defining your.</p>
			<h2>Examples</h2>
			<IconsProvider />
			<List {...props} />
		</div>
	))
	.add('large', () => {
		const eprops = Object.assign({}, props);
		eprops.displayMode = 'large';
		return (
			<div>
				<h1>List</h1>
				<p>Display the list in large mode</p>
				<IconsProvider />
				<List {...eprops} />
			</div>
		);
	})
	.add('tile', () => {
		const tprops = Object.assign({}, props);
		tprops.displayMode = 'tile';
		return (
			<div>
				<h1>List</h1>
				<p>Display the list in tile mode</p>
				<IconsProvider />
				<List {...tprops} />
			</div>
		);
	});
