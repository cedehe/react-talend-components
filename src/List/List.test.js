import React from 'react';
import renderer from 'react-test-renderer';

import List from './List.component';

jest.mock('react-dom');

const props = {
	displayMode: 'table',
	list: {
		columns: [
			{ key: 'id', label: 'Id' },
			{ key: 'name', label: 'Name' },
		],
		items: [
			{ id: 1, name: 'Hello world' },
			{ id: 2, name: 'Foo' },
			{ id: 3, name: 'Bar' },
		],
		titleProps: {
			onClick: jest.fn(),
		},
	},
	toolbar: {
		onClickAdd: jest.fn(),
		onFilter: jest.fn(),
		onSelectSortBy: jest.fn(),
		onSelectDisplayMode: jest.fn(),
		sortBy: [
			{ name: 'Id' },
			{ name: 'Name', selected: true },
		],
	},
};

describe('List', () => {
	it('should render its name', () => {
		// when
		const wrapper = renderer.create(<List {...props} />).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render id if provided', () => {
		// given
		const tProps = {
			id: 'context',
			...props,
		};

		// when
		const wrapper = renderer.create(<List {...tProps} />).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});
});
