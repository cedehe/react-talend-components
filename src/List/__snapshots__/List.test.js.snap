exports[`List should not render the toolbar without toolbar props 1`] = `
<div
  className="tc-list">
  <div>
    <ul
      className="tc-list-items table">
      <li
        className="tc-list-header">
        <div
          className="tc-list-header-item tc-list-header-text">
          Id
        </div>
        <div
          className="tc-list-header-item tc-list-header-text">
          Name
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={1}>
                1
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Hello world">
                Hello world
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={2}>
                2
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Foo">
                Foo
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={3}>
                3
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Bar">
                Bar
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`List should render 1`] = `
<div
  className="tc-list">
  <div>
    <div
      className="navbar navbar-default"
      role="toolbar">
      <div
        className="container-fluid">
        <form
          className="navbar-form navbar-left">
          <div
            className="navbar-text">
            <input
              checked={false}
              disabled={false}
              id={undefined}
              onChange={[Function]}
              type="checkbox" />
            <label
              className="tc-list-toolbar-select-all"
              htmlFor={undefined}>
              Select All
            </label>
          </div>
        </form>
        <p
          className="navbar-text">
          <label
            htmlFor={undefined}>
            Display:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id={42}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby={42}
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id={undefined}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-bars"
                    title={null} />
                  Table
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id={undefined}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-list-alt"
                    title={null} />
                  Expanded
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id={undefined}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-th-large"
                    title={null} />
                  Tile
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <p
          className="navbar-text">
          <label
            htmlFor={undefined}>
            Sort by:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id={42}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              Name
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby={42}
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id={undefined}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  Id
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id={undefined}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  Name
                </a>
              </li>
            </ul>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              role="button">
              ASCENDING
            </a>
          </li>
        </ul>
        <p
          className="navbar-text">
          <label
            htmlFor={undefined}>
            Show:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id={42}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              5
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby={42}
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  5
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  10
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  20
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  50
                </a>
              </li>
            </ul>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-backward"
                title={null} />
            </a>
          </li>
          <li
            className="btn-link tc-pagination-ctrl-prev"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-play"
                title={null} />
            </a>
          </li>
          <li
            className="disabled"
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}>
              <span
                className="btn-link">
                2
                /
                3
              </span>
            </a>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-play"
                title={null} />
            </a>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-forward"
                title={null} />
            </a>
          </li>
        </ul>
        <form
          className="navbar-form navbar-right"
          onSubmit={[Function]}
          role="search">
          <div
            className="form-group">
            <input
              aria-label="Filter"
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Filter"
              type="search" />
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              title="Filter"
              type="submit">
              <svg
                aria-hidden="true"
                className="tc-svg-icon"
                title={null}>
                <use
                  xlinkHref="#talend-search" />
              </svg>
              <span
                className="sr-only">
                Filter
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div>
    <ul
      className="tc-list-items table">
      <li
        className="tc-list-header">
        <div
          className="tc-list-header-item tc-list-header-text">
          Id
        </div>
        <div
          className="tc-list-header-item tc-list-header-text">
          Name
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={1}>
                1
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Hello world">
                Hello world
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={2}>
                2
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Foo">
                Foo
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id={undefined}
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id={undefined}
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id={undefined}
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={3}>
                3
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Bar">
                Bar
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`List should render id if provided 1`] = `
<div
  className="tc-list">
  <div>
    <div
      className="navbar navbar-default"
      role="toolbar">
      <div
        className="container-fluid">
        <form
          className="navbar-form navbar-left">
          <div
            className="navbar-text">
            <input
              checked={false}
              disabled={false}
              id="context-check-all"
              onChange={[Function]}
              type="checkbox" />
            <label
              className="tc-list-toolbar-select-all"
              htmlFor="context-check-all">
              Select All
            </label>
          </div>
        </form>
        <p
          className="navbar-text">
          <label
            htmlFor="context-display-mode">
            Display:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id="context-display-mode"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby="context-display-mode"
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id="context-display-mode-table"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-bars"
                    title={null} />
                  Table
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id="context-display-mode-large"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-list-alt"
                    title={null} />
                  Expanded
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id="context-display-mode-tile"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  <i
                    aria-hidden="true"
                    className="fa fa-th-large"
                    title={null} />
                  Tile
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <p
          className="navbar-text">
          <label
            htmlFor="context-sort-by">
            Sort by:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id="context-sort-by"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              Name
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby="context-sort-by"
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id="context-sort-by-item-id"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  Id
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  id="context-sort-by-item-name"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  Name
                </a>
              </li>
            </ul>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="context-sort-order"
              onClick={[Function]}
              role="button">
              ASCENDING
            </a>
          </li>
        </ul>
        <p
          className="navbar-text">
          <label
            htmlFor="context-pagination-size">
            Show:
          </label>
        </p>
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className="dropdown"
            style={undefined}>
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href=""
              id="context-pagination-size"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button">
              5
               
              <span
                className="caret" />
            </a>
            <ul
              aria-labelledby="context-pagination-size"
              className="dropdown-menu"
              role="menu">
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  5
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  10
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  20
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}>
                <a
                  href=""
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1">
                  50
                </a>
              </li>
            </ul>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="context-pagination-nav-to-first"
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-backward"
                title={null} />
            </a>
          </li>
          <li
            className="btn-link tc-pagination-ctrl-prev"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="context-pagination-nav-to-prev"
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-play"
                title={null} />
            </a>
          </li>
          <li
            className="disabled"
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}>
              <span
                className="btn-link">
                2
                /
                3
              </span>
            </a>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="context-pagination-nav-to-next"
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-play"
                title={null} />
            </a>
          </li>
          <li
            className="btn-link"
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="context-pagination-nav-to-last"
              onClick={[Function]}
              role="button">
              <i
                aria-hidden="true"
                className="fa fa-forward"
                title={null} />
            </a>
          </li>
        </ul>
        <form
          className="navbar-form navbar-right"
          onSubmit={[Function]}
          role="search">
          <div
            className="form-group">
            <input
              aria-label="Filter"
              className="form-control"
              id="context-filter-input"
              onChange={[Function]}
              placeholder="Filter"
              type="search" />
            <button
              className="btn btn-link"
              disabled={false}
              id="context-filter-submit"
              title="Filter"
              type="submit">
              <svg
                aria-hidden="true"
                className="tc-svg-icon"
                title={null}>
                <use
                  xlinkHref="#talend-search" />
              </svg>
              <span
                className="sr-only">
                Filter
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div>
    <ul
      className="tc-list-items table">
      <li
        className="tc-list-header">
        <div
          className="tc-list-header-item tc-list-header-text">
          Id
        </div>
        <div
          className="tc-list-header-item tc-list-header-text">
          Name
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id="context-0"
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id="context-0-title"
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id="context-0-check"
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={1}>
                1
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Hello world">
                Hello world
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id="context-1"
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id="context-1-title"
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id="context-1-check"
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={2}>
                2
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Foo">
                Foo
              </div>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="tc-list-item"
          id="context-2"
          onClick={undefined}
          onDoubleClick={[Function]}
          role="button">
          <div
            className="tc-list-item-title">
            <button
              className="btn btn-link"
              disabled={false}
              id="context-2-title"
              onClick={[Function]}
              role="link"
              title={undefined}
              type="button" />
          </div>
          <input
            checked={false}
            id="context-2-check"
            onChange={[Function]}
            type="checkbox" />
          <div
            className="tc-list-item-definition-list">
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Id
              </div>
              <div
                className="tc-list-item-definition-description"
                title={3}>
                3
              </div>
            </div>
            <div
              className="tc-list-item-definition-item">
              <div
                className="tc-list-item-definition-term">
                Name
              </div>
              <div
                className="tc-list-item-definition-description"
                title="Bar">
                Bar
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
